contracts/JBPaymentTerminal.sol

```diff
@@ -0,0 +1,987 @@
+    {
+      // Get the amount of discount that should be applied to any fees take.
+      uint256 _feeDiscount = _getFeeDiscount(_projectId);
+
+      uint256 _feeEligibleDistributionAmount;
+
+      // Payout to splits and get a reference to the leftover transfer amount after all mods have been paid.
+      // Also get a reference to the amount that was distributed that is eligible to have fees taken.
+      // The net transfer amount is the withdrawn amount minus the fee.
+      (_leftoverDistributionAmount, _feeEligibleDistributionAmount) = _distributeToPayoutSplitsOf(
+        _projectId,
+        _fundingCycle,
+        _distributedAmount,
+        _feeDiscount
+      );
+
+      // Leftover distribution amount is also eligible for a fee since the funds are going out of the ecosystem to _beneficiary.
+      _feeEligibleDistributionAmount += _leftoverDistributionAmount;
+
+      // Take the fee.
+      _feeAmount = _projectId == 1 || _feeEligibleDistributionAmount == 0 || fee == 0
+        ? 0
+        : _takeFeeFrom(
+          _projectId,
+          _fundingCycle,
+          _feeEligibleDistributionAmount,
+          _projectOwner,
+          _feeDiscount
+        );
+
+      // Transfer any remaining balance to the project owner.
+      if (_leftoverDistributionAmount > 0)
+        _transferFrom(
+          address(this),
+          _projectOwner,
+          _leftoverDistributionAmount - _getFeeAmount(_leftoverDistributionAmount, _feeDiscount)
+        );
+    }
+
+    emit DistributePayouts(
+      _fundingCycle.configuration,
+      _fundingCycle.number,
+      _projectId,
+      _projectOwner,
+      _amount,
+      _distributedAmount,
+      _feeAmount,
+      _leftoverDistributionAmount,
+      _memo,
+      msg.sender
+    );
+  }
+
+  /**
+    @notice
+    Allows a project to send funds from its overflow up to the preconfigured allowance.
+
+    @dev
+    Only a project's owner or a designated operator can migrate it.
+
+    @param _projectId The ID of the project to use the allowance of.
+    @param _amount The amount of the allowance to use.
+    @param _beneficiary The address to send the funds to.
+  */
+  function useAllowanceOf(
+    uint256 _projectId,
+    uint256 _amount,
+    uint256 _currency,
+    uint256 _minReturnedWei,
+    address payable _beneficiary
+  )
+    external
+    override
+    nonReentrant
+    requirePermission(projects.ownerOf(_projectId), _projectId, JBOperations.USE_ALLOWANCE)
+  {
+    // Record the use of the allowance.
+    (JBFundingCycle memory _fundingCycle, uint256 _withdrawnAmount) = store.recordUsedAllowanceOf(
+      _projectId,
+      _amount,
+      _currency,
+      _minReturnedWei
+    );
+
+    // Get a reference to the project owner, which will receive tokens from paying the platform fee
+    // and receive any extra distributable funds not allocated to payout splits.
+    address payable _projectOwner = payable(projects.ownerOf(_projectId));
```

Comment:
payable is not needed anymore

Reply:
true. seems like we haven't needed it in a while